tests/integration_tests/download_cache.py

```diff
@@ -32,6 +32,24 @@ def download_and_apply_cache(name: str, cache_directory: str | None = None, vers
         raise Exception(f'Failed to download cache from {uri} and extract to {cache_directory}.') from exc
 
 
+def download_taxonomy_package(name: str, download_path: str, version_id: str | None = None) -> None:
+    """
+    :param name: Filename (including extension) of taxonomy package to download
+    :param download_path: Location to save the package
+    :param version_id: The S3 object version to retrieve. None for latest.
+    """
+    if os.path.exists(download_path):
+        return
+    uri = get_s3_uri(
+        f'ci/taxonomy_packages/{name}',
+        version_id=version_id
+    )
+    try:
+        urllib.request.urlretrieve(uri, download_path)
```

Comment:
Is there a race condition here for test suites with common packages that execute in multiple processes on the same node?

Reply:
This is called before the shard mechanism introduces parallelism, so I can't think of any race condition unless we execute the conformance suite runner on two completely separate processes on the same node.

Reply:
I do this all the time locally 😅 , but I guess it's not a huge issue since you only need to download the packages once locally